### PR TITLE
Add RFC template draft and README file

### PR DESCRIPTION
**Pull request by cavalheiro**
### What does this PR do?

This PR adds a template and README file for the Salt RFC process proposal. The "RFC" (request for comments) process is intended to provide a consistent and controlled path for new features to enter the project.

This process is being **actively developed**, and it will still change as more features are implemented and the community settles on specific approaches to feature development.

### Tests written?

Not applicable

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by rallytime** [comment]
@cavalheiro Thank you! This is a great start! I am really pleased with this idea.

@saltstack/team-core Can you all take a look at this here?

**Review by twangboy — APPROVED** [review]

**Comment by cavalheiro** [comment]
What should be the next steps?

**Comment by rallytime** [comment]
I think this looks like a great start. I'd like some more members of the core team to look at this, so I have tagged them for a review. I ❤️ this idea and am excited to get it in. Thank you so much for submitting.

File: rfcs/0000-template.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+- Feature Name: (fill with a unique ident, my_awesome_feature)
```
Comment (rallytime) [inline rfcs/0000-template.md]: Let's expand `ident` to `identifier` or `identity`.

**Review by rallytime — COMMENTED** [review]
I have one _very_ small comment.

Additionally, I think a mention in the RFC template about potential test cases or a testing plan for automated tests would be good as well.

**Review by gtmanfred — APPROVED** [review]
i really like this.

**Review by thatch45 — APPROVED** [review]

**Comment by cavalheiro** [comment]
@rallytime, added a commit with the changes you suggested.
Thank you for the feedback!

**Review by terminalmage — APPROVED** [review]

**Comment by cachedout** [comment]
We may wish to also edit the PR template itself to reference this. Otherwise, this is great!

**Review by cachedout — APPROVED** [review]

**Comment by cavalheiro** [comment]
Added RFC life cycle description to README file

**Review by rallytime — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Comment by cachedout** [comment]
@rallytime This looks good. Let's get this rolling!

**Pull request merged**